src/PassEdit.php

```diff
@@ -1,28 +1,47 @@
 <?php
 
-use MediaWiki\MediaWikiServices;
+use MediaWiki\User\UserFactory;
+use Wikimedia\Rdbms\ILoadBalancer;
 
 class PassEdit extends SpecialPage {
-    function __construct() {
-		parent::__construct('EditPassword', 'editpassword');
-	}
+    private PasswordFactory $passwordFactory;
+    private ILoadBalancer $loadBalancer;
+    private UserFactory $userFactory;
+
+    function __construct(PasswordFactory $passwordFactory, ILoadBalancer $loadBalancer, UserFactory $userFactory) {
+        parent::__construct('EditPassword', 'editpassword');
+        $this->passwordFactory = $passwordFactory;
+        $this->loadBalancer = $loadBalancer;
+        $this->userFactory = $userFactory;
+    }
 
     function getGroupName() {
         return 'users';
     }
 
+    function setCSRFToken($session) {
+        $session->persist();
+        $csrftoken = $session->getToken();
+        $session->save();
+        return $csrftoken;
+    }
+
+    function isCSRF($session, $csrftoken) {
```

Comment:
Took me some time to realize that this checks whether there is cross-site request forgery, not whether the token is valid. Can this be renamed to something like `isCSRFTokenInvalid`? Not every invalid token is because of CSRF.